icalevents/icalparser.py

```diff
@@ -255,16 +276,110 @@ def parse_events(content, start=None, end=None, default_span=timedelta(days=7)):
+                    if str(e.start.tzinfo) in timezones:
+                        start_tz = timezones[str(e.start.tzinfo)]
+                    else:
+                        try:
+                            start_tz = timezone(str(e.start.tzinfo))
+                        except:
+                            pass
+
+                if e.end.tzinfo != UTC:
+                    if str(e.end.tzinfo) in timezones:
+                        end_tz = timezones[str(e.end.tzinfo)]
+                    else:
+                        try:
+                            end_tz = timezone(str(e.end.tzinfo))
+                        except:
+                            pass
+
+            # If we've been passed or constructed start/end values
+            # that are timezone naive, but the actual appointment
+            # start and end times are in a timezone, convert start
+            # and end to have a timezone. Otherwise, python will
+            # raise an exception for comparing timezone naive
+            # and offset-aware values.
+            if e.start.tzinfo and not start.tzinfo:
+                start = normalize(start, e.start.tzinfo)
+            if e.start.tzinfo and not end.tzinfo:
+                end = normalize(end, e.start.tzinfo)
+
+            duration = e.end - e.start
             if e.recurring:
                 # Unfold recurring events according to their rrule
                 rule = parse_rrule(component, cal_tz)
                 dur = e.end - e.start
-                found.extend(e.copy_to(dt) for dt in rule.between(start - dur, end, inc=True))
+                after = start - dur
+
+                for dt in rule.between(after, end, inc=True):
+                    if start_tz is None:
+                        # Shrug. If we coudln't work out the timezone, it is what it is.
+                        ecopy = e.copy_to(dt, e.uid)
+                    else:
+                        # Recompute the start time in the current timezone *on* the
+                        # date of *this* occurrence. This handles the case where the
+                        # recurrence has crossed over the daylight savings time boundary.
+                        naive = datetime(dt.year, dt.month, dt.day, dt.hour, dt.minute, dt.second)
+                        dtstart = start_tz.localize(naive)
+
+                        ecopy = e.copy_to(dtstart, e.uid)
+
+                        # We're effectively looping over the start time, we might need
+                        # to adjust the end time too, but don't have it's recurred value.
+                        # Make sure it's adjusted by constructing it from the meeting
+                        # duration. Pro: it'll be right. Con: if it was in a different
+                        # timezone from the start time, we'll have lost that.
+                        ecopy.end = dtstart + duration
+
+                    exdate = "%04d%02d%02d" % (ecopy.start.year, ecopy.start.month, ecopy.start.day)
```

Comment:
Hmm, not really a question for this PR but maybe more for the project as a whole but wouldn't it make sense to use the more and more readably `f"…"`-strings or `"…".format(…)`-strings if we want to support python 3.5 and prior versions? 

Python 3.5 is EOL later this year https://devguide.python.org/#status-of-python-branches 
If supporting it is still a priority we could use `.format()`, if not I'd go with f-strings.

Reply:
I don't feel strongly about it; but yes, doing it the same way throughout probably makes sense.